Utilize std::make_unsigned_t instead of wrapping lshift

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35422 Make intdiv_256 a more generic binary operator template
* #35339 lshift and rshift on CUDA should match the behavior on CPU
* **#35323 Utilize std::make_unsigned_t instead of wrapping lshift**

Currently lshift_kernel uses a hacky wrapper that wraps all integer
types. This is unnecessary because we have std::make_unsigned_t

Differential Revision: [D20824641](https://our.internmc.facebook.com/intern/diff/D20824641)

xuhdev committed:
Utilize std::make_unsigned_t instead of wrapping lshift

Currently lshift_kernel uses a hacky wrapper that wraps all integer
types. This is unnecessary because we have std::make_unsigned_t

[ghstack-poisoned]

xuhdev committed:
Update on "Utilize std::make_unsigned_t instead of wrapping lshift"


Currently lshift_kernel uses a hacky wrapper that wraps all integer
types. This is unnecessary because we have std::make_unsigned_t

[ghstack-poisoned]

zou3519 reviewed (approved):
nice!

zou3519 commented on aten/src/ATen/native/cpu/BinaryOpsKernel.cpp:
```diff
@@ -186,27 +186,6 @@ void bitwise_xor_kernel(TensorIterator& iter) {
-}
-
-static inline int64_t lshift_wrapper(int64_t a, int64_t b) {
-  return ((uint64_t)a) << b;
```
Just out of curiosity, why do we have to make the left hand side unsigned before performing the shift?

xuhdev replied:
Probably because shifting on signed integers in C++ is implementation defined. See https://stackoverflow.com/a/4009954/1150462 for an explanation and reference.

xuhdev committed:
Update on "Utilize std::make_unsigned_t instead of wrapping lshift"


Currently lshift_kernel uses a hacky wrapper that wraps all integer
types. This is unnecessary because we have std::make_unsigned_t

[ghstack-poisoned]

xuhdev commented:
Weird, this has actually been merged in beac3f27f0ef9340188ac645b108e7ba32faafe0 . Manually closing now.
